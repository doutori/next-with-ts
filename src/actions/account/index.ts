import { GET_ACCOUNT, UPDATE_ACCOUNT } from '~/constant';
import { AccountActionTypes } from '~/actionTypes';
import { Account } from '~/modelTypes';

export const getAccount = (): AccountActionTypes => ({
    type: GET_ACCOUNT
});

export const updateAccount = (account: Account): AccountActionTypes => ({
    type: UPDATE_ACCOUNT,
    payload: { data: account }
});
