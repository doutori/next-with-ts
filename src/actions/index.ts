import { INCREMENT, DECREMENT } from '../constant';
import { CounterActionTypes } from '../actionTypes';

export const increment = (number: number): CounterActionTypes => ({
    type: INCREMENT,
    payload: number
});

export const decrement = (number: number): CounterActionTypes => ({
    type: DECREMENT,
    payload: number
});
