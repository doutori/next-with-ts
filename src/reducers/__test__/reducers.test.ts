/* eslint-env jest */
import snapshotDiff from 'snapshot-diff';
import { counterActions } from '../../actions';

import { counter, initialState } from '../';

describe('Counter reducer', () => {
    it('increment testing', () => {
        expect(
            snapshotDiff(
                initialState,
                counter(initialState, counterActions.increment(1))
            )
        ).toMatchSnapshot();
    });

    it('increment testing', () => {
        expect(
            snapshotDiff(
                initialState,
                counter(initialState, counterActions.decrement(1))
            )
        ).toMatchSnapshot();
    });
});
