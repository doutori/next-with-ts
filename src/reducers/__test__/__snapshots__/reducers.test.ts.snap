// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter reducer increment testing 1`] = `
"Snapshot Diff:
- First value
+ Second value

  Object {
-   \\"count\\": 0,
+   \\"count\\": 1,
  }"
`;

exports[`Counter reducer increment testing 2`] = `
"Snapshot Diff:
- First value
+ Second value

  Object {
-   \\"count\\": 0,
+   \\"count\\": -1,
  }"
`;
